tasks/docs/serve.js

```diff
@@ -27,7 +28,8 @@ var
 
 module.exports = function () {
     // use a different config
-    config = configSetup.addDerivedValues(config);
+    const config = extend(true, {}, configDocs);
```

Comment:
This is still a confusing change. Please revert. addDerivedValues returns config and config is declared via let, so what's the purpose of creating a temporary const which extends the previous config.... This is confusing.

Reply:
`addDerivedValues()` was written like when a config was given, the config was a) mutated, b) returned

now it keeps mutating it (as you would expect by the function name) and we extend/clone the config from import (before mutation) correctly

with this explanation, is it fine?

Reply:
`addDerivedValues()` was written quite complited. When no/empty config was given, it created it. Why? When such fallback is removed, no need for return.

By the function name I would expect a void method that adds some config values. This is how it is coded now, is that ok?

The change also fixes the mutation of a config returned from an imported module, previously, it was not extended/cloned. IMHO it is a good practise to never mutate anything imported.

Reply:
It is still confusing: Why is configDocs now declared outiside of the exports scope? For nothing else than being extended for the const once.... Why not move the `let config      = require('../config/docs'),` into the module.exports block and keep the functionality as it was before?

Reply:
because it is a good practise to keep imports at the top (and never mutate them) - otherwise eslint with recommended config will warn - otherwise we will need `eslint-disable-line`

Reply:
ok, understood, but then it should be kept as var instead if let if we get into scrop issues otherwise ;)

Reply:
If imported on top (with var/let/const), then it is visible to the whole script scope.

Reply:
now part of #2597

see https://github.com/fomantic/Fomantic-UI/pull/2597/commits/117dc67a0369bc3d5f70a747081261eb03856825, we import the config as `configDocs`, then create (local/cloned) `config` config.